src/index.ts

```diff
@@ -0,0 +1,508 @@
+  _requestAnimationFrame(callback: () => void) {
+    let fired = false;
+    const interval = setInterval(() => {
+      if (!fired && document.hidden) {
+        fired = true;
+        clearInterval(interval);
+        callback();
+      }
+    }, 1000 / this.fps);
+    requestAnimationFrame(() => {
+      if (!fired) {
+        fired = true;
+        clearInterval(interval);
+        callback();
+      }
+    });
+  }
+
+  start() {
+
+    // Hidden canvas element for merging
+    const canvas = this._canvas = document.createElement('canvas');
+    this._canvas.setAttribute('width', this.width.toString());
+    this._canvas.setAttribute('height', this.height.toString());
+    this._canvas.setAttribute('style', 'position:fixed; left: 110%; pointer-events: none'); // Push off screen
+    this._ctx = this._canvas.getContext('2d');
+
+    this.started = true;
+    this._requestAnimationFrame(this._draw.bind(this));
+
+    // Add video
+    this.result = this._canvas?.captureStream(this.fps) || null;
+
+    // Remove "dead" audio track
+    const deadTrack = this.result?.getAudioTracks()[0];
+    if (deadTrack) {this.result?.removeTrack(deadTrack);}
+
+    // Add audio
+    const audioTracks = this._audioDestination?.stream.getAudioTracks();
+    if (audioTracks && audioTracks.length) {
+      this.result?.addTrack(audioTracks[0]);
+    }
+  }
+
+  _updateAudioDelay(delayInMs: number) {
+    if (this._videoSyncDelayNode && this._audioCtx) {
+      this._videoSyncDelayNode.delayTime.setValueAtTime(delayInMs / 1000, this._audioCtx.currentTime);
+    }
+  }
+
+  _draw() {
+    if (!this.started) {return;}
+
+    this._frameCount++;
+
+    // update video processing delay every 60 frames
+    let t0  = 0;
+    if (this._frameCount % 60 === 0) {
+      t0 = performance.now();
+    }
+
+    let awaiting = this._streams.length;
+    const done = () => {
+      awaiting--;
+      if (awaiting <= 0) {
+        if (this._frameCount % 60 === 0) {
+          const t1 = performance.now();
+          this._updateAudioDelay(t1 - t0);
+        }
+        this._requestAnimationFrame(this._draw.bind(this));
+      }
+    };
+
+    if (this.clearRect) {
+      this._ctx?.clearRect(0, 0, this.width, this.height);
+    }
+    this._streams.forEach((stream) => {
+      if (stream.draw) { // custom frame transform
+        stream.draw(this._ctx, stream.element, done);
+      } else if (!stream.isData && stream.hasVideo) {
+        this._drawVideo(stream.element, stream);
+        done();
+      } else {
+        done();
+      }
+    });
+
+    if (this._streams.length === 0) {
+      done();
+    }
+  }
+
+  _drawVideo(element: HTMLVideoElement, stream: any) {
+
+
+    // default draw function
+    const canvasSize = { height: this.height, width: this.width};
+
+    const position = {
+      x: stream.x || 0,
+      y: stream.y || 0
+    };
+
+    const size = {
+        height: stream.height || element.videoHeight || canvasSize.height,
```

Comment:
Using the videoWidth over canvas.width is new behaviour - is this intentional?

Reply:
This is in order to get the actual video size in order to keep the ratio.
We may only do that if keepRatio is true, if you d'like.

Reply:
I agree aspect ratio should be kept, but not specifying a width and height should still fullscreen the stream.

Revert this for now - will address in another PR.